mlrun/serving/routers.py

```diff
@@ -157,3 +163,256 @@ def _handle_event(self, event):
+        self,
+        context,
+        name,
+        routes=None,
+        protocol=None,
+        url_prefix=None,
+        health_prefix=None,
+        vote_type=None,
+        executor_type=None,
+        **kwargs,
+    ):
+        super().__init__(
+            context, name, routes, protocol, url_prefix, health_prefix, **kwargs
+        )
+        self.url_prefix = url_prefix or f"/{self.protocol}/models"
+        self.name = "VoteRouter"
+        self.vote_type = vote_type
+        self.vote_flag = True if self.vote_type is not None else False
+        self.executor_type = executor_type
+        self._model_logger = _ModelLogPusher(self, context)
+        self.version = kwargs.get("version", "v1")
+        self.log_router = True
+
+    def _resolve_route(self, body, urlpath):
+        """Resolves the appropriate model to send the event to.
+        Supports:
+        - <prefix>/<model>[/versions/<ver>]/operation
+        Sends the event to the specific <model>[/versions/<ver>]
+
+        - <prefix>/operation
+        Sends the event to all models
+
+        Args:
+            body (dict): event body
+            urlpath (string): url path
+
+        Raises:
+            ValueError: [description]
+
+        Returns:
+            [type]: [description]
+        """
+        subpath = None
+        model = ""
+        if urlpath and not urlpath == "/":
+            # process the url <prefix>/<model>[/versions/<ver>]/operation
+            subpath = ""
+            urlpath = urlpath[len(self.url_prefix) :].strip("/")
+            if not urlpath:
+                # Only self.url_prefix was given
+                return "", None, ""
+            segments = urlpath.split("/")
+            if len(segments) == 1:
+                # Are we looking at a router level operation?
+                if segments[0] in ["infer", "predict", "explain"]:
+                    # We are given an operation
+                    return self.name, None, segments[0]
+            model = segments[0]
+            if len(segments) > 2 and segments[1] == "versions":
+                model = model + ":" + segments[2]
+                segments = segments[2:]
+            if len(segments) > 1:
+                subpath = "/".join(segments[1:])
+
+        if isinstance(body, dict):
+            # accepting route information from body as well
+            # to support streaming protocols (e.g. Kafka).
+            model = model or self.name
+            subpath = body.get("operation", subpath)
+        if subpath is None:
+            subpath = "infer"
+
+        if model in self.routes:
+            self.log_router = False
+        elif model != self.name:
+            models = " | ".join(self.routes.keys())
+            raise ValueError(
+                f"model {model} doesnt exist, available models: {models} or an operation alone for ensemble operation"
+            )
+        else:
+            self.log_router = True
+        return model, self.routes[model], subpath
```

Comment:
`self.routes[model]` will raise exception if `model` is `self.name` no ?
did you test this code ?

Reply:
If it's a router level call we return earlier, only specific model calls get this far. 
and yes, this was tested locally and after deployment incl. tracking